tests/model_registry/rest_api/conftest.py

```diff
@@ -63,3 +92,250 @@ def updated_model_registry_resource(
+    Args:
+        model_registry_db_secret: The secret for the model registry's MySQL database
+    Returns:
+        dict[str, Any]: The patched MySQL template
+    """
+    mysql_template = get_model_registry_deployment_template_dict(
+        secret_name=model_registry_db_secret.name,
+        resource_name=DB_RESOURCES_NAME,
+    )
+    mysql_template["spec"]["containers"][0]["args"].append(f"--ssl-ca={CA_FILE_PATH}")
+    mysql_template["spec"]["containers"][0]["volumeMounts"].append({
+        "mountPath": CA_MOUNT_PATH,
+        "name": CA_CONFIGMAP_NAME,
+        "readOnly": True,
+    })
+    mysql_template["spec"]["volumes"].append({"name": CA_CONFIGMAP_NAME, "configMap": {"name": CA_CONFIGMAP_NAME}})
+    return mysql_template
+
+
+@pytest.fixture(scope="class")
+def deploy_secure_mysql_and_mr(
+    model_registry_namespace: str,
+    model_registry_db_secret: Secret,
+    model_registry_db_deployment: Deployment,
+    model_registry_mysql_config: dict[str, Any],
+    mysql_template_with_ca: dict[str, Any],
+    patch_mysql_deployment_with_ssl_ca: Deployment,
+) -> Generator[ModelRegistry, None, None]:
+    """
+    Deploy a secure MySQL and Model Registry instance.
+    Args:
+        model_registry_namespace: The namespace of the model registry
+        model_registry_db_secret: The secret for the model registry's MySQL database
+        model_registry_db_deployment: The deployment for the model registry's MySQL database
+        model_registry_mysql_config: The MySQL config dictionary
+        mysql_template_with_ca: The MySQL template with the CA file path and volume mount
+    """
+    with ModelRegistry(
+        name=SECURE_MR_NAME,
+        namespace=model_registry_namespace,
+        label=MODEL_REGISTRY_STANDARD_LABELS,
+        grpc={},
+        rest={},
+        istio=ISTIO_CONFIG_DICT,
+        mysql=model_registry_mysql_config,
+        wait_for_resource=True,
+    ) as mr:
+        mr.wait_for_condition(condition="Available", status="True")
+        yield mr
+
+
+@pytest.fixture()
+def local_ca_bundle(request: pytest.FixtureRequest, admin_client: DynamicClient) -> Generator[str, Any, Any]:
+    """
+    Creates a local CA bundle file by fetching the CA bundle from a ConfigMap and appending the router CA from a Secret.
+    Args:
+        request: The pytest request object
+        admin_client: The admin client to get the CA bundle from a ConfigMap and append the router CA from a Secret.
+    Returns:
+        Generator[str, Any, Any]: A generator that yields the CA bundle path.
+    """
+    ca_bundle_path = getattr(request, "param", {}).get("cert_name", "ca-bundle.crt")
+    create_ca_bundle_file(
+        admin_client=admin_client,
+        namespace=py_config["model_registry_namespace"],
+        ca_bundle_path=ca_bundle_path,
+        cert_name=ca_bundle_path,
+    )
+    yield ca_bundle_path
+
+    os.remove(ca_bundle_path)
+
+
+@pytest.fixture(scope="class")
+def test_ca_configmap(
```

Comment:
We should not name anything starting with `test`. May be rename it to ca_configmap_for_test or something similar.